packages/core/src/services/ai/index.ts

```diff
@@ -146,13 +155,29 @@ export async function ai({
     const useSchema = schema && !!output && output !== 'no-schema'
     const resultType: StreamType = useSchema ? 'object' : 'text'
 
+    const stopWhen = getStopWhen({ maxSteps: config.maxSteps })
     const result = streamText({
       ...omit(config, ['schema']),
+      ...stopWhen,
+      messages,
+      onError: ({ error }) => {
+        const span = tracer.scope().active()
+        if (!span) {
+          console.error(error)
+          return
+        }
+
+        span.log({
+          event: '[Latitude]: AI Provider Call Error',
+          message: (error as Error)?.message,
+          error,
+        })
```

Comment:
I think we can remove these traces @geclos this is a handle error. What I saw is if we throw this error before reading the failed promise reponses we avoid the error

Reply:
let's talk it tomorrow offline